functional/sev/run.sh

```diff
@@ -0,0 +1,444 @@
+#!/bin/bash
+# Copyright 2022 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -eE
+set -o nounset
+set -o pipefail
+
+trap cleanup EXIT
+
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+NC='\033[0m'
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+tests_repo_dir="$(realpath "$script_dir/../..")"
+
+export WORKING_DIR="${HOME}/work"
+
+export EDK2_REPO_DIR="${HOME}/edk2"
+export DOCKER_USER="rksavino"
+
+export GOROOT="/usr/local/go"
+export GOPATH="${WORKING_DIR}/go"
+export PATH="${PATH}:${GOROOT}/bin"
+
+export KBS_DB_HOST="127.0.0.1"
+export KBS_DB_USER="root"
+export KBS_DB_PW="root"
+export KBS_DB="simple_kbs"
+export KBS_DB_TYPE="mysql"
+
+esudo() {
+  sudo -E PATH=$PATH "$@"
+}
+
+build_and_install_kata_runtime_4270() {
+  cd ${WORKING_DIR}
+
+  # Clone kata-containers repo and checkout PR 4270
+  git clone https://github.com/kata-containers/kata-containers.git
+  cd kata-containers
+  gh pr checkout 4270
+
+  # Build and install the CC Kata Containers runtime
+  export GO111MODULE=on
+  cd ${WORKING_DIR}/kata-containers/src/runtime
+  make && esudo make install
+}
+
+build_rootfs() {
+  cd ${WORKING_DIR}/kata-containers/tools/osbuilder/rootfs-builder
+  esudo GOPATH=$GOPATH AGENT_INIT=yes USE_DOCKER=true DOCKER_RUNTIME=runc AA_KBC=offline_sev_kbc SKOPEO_UMOCI=yes ./rootfs.sh ubuntu
```

Comment:
At the point this script is executed, the whole kata-containers was built and installed. So maybe we should not rebuild the image here....

The Jenkins job will export `CI_JOB=CC_SEV_CRI_CONTAINERD_K8S` and the variables on the block https://github.com/kata-containers/tests/blob/ab84d51927209ab9df8f144b63fbaaec8f8ef6c8/.ci/ci_job_flags.sh#L111 are all exported afterwards. The list of variables are:
```
KATA_BUILD_CC="yes"
CCV0="yes"
UMOCI="yes"
AA_KBC="offline_fs_kbc"
KUBERNETES=yes
TEE_TYPE="sev"
```
Ok, the value of `AA_KBC` is currently `offline_fs_kbc` but in this script it is `offline_sev_kbc`. We probably should adjust this. Something like that:
```
vagrant@ubuntu2004:~/go/src/github.com/kata-containers/tests$ git diff
diff --git a/.ci/ci_job_flags.sh b/.ci/ci_job_flags.sh
index 946704ed..be9c11e9 100755
--- a/.ci/ci_job_flags.sh
+++ b/.ci/ci_job_flags.sh
@@ -117,7 +117,11 @@ case "${CI_JOB}" in
                                export SKOPEO=yes
                        fi
                        if [[ "${CI_JOB}" =~ K8S ]]; then
-                               export AA_KBC="offline_fs_kbc"
+                               if [[ "${CI_JOB}" =~ SEV ]]; then
+                                       export AA_KBC="offline_sev_kbc"
+                               else
+                                       export AA_KBC="offline_fs_kbc"
+                               fi
                                export KUBERNETES=yes
                        fi
                        if [ "${CI_JOB}" == "CC_SEV_CRI_CONTAINERD_K8S" ]; then

```
For sev case, do we need skopeo in the guest image? I thought the plan was to release with image-rs support instead. I am asking to assess if `SKOPEO_UMOCI=yes` should really be exported. Actually it seems `SKOPEO_UMOCI` isn't even used by the build scripts.

Reply:
SEV - don't need SKOPEO. Adjusted `ci_job_flags.sh` to export additional SEV params for `KATA_BUILD_KERNEL_TYPE`, `AA_KBC`, and `TEST_INITRD`.